documentation.md

```diff
@@ -0,0 +1,669 @@
+
+<figure class="image">
+  <img src="https://i.imgur.com/H1FLUsu.png" alt="Delay calculator wrapper">
+  <figcaption>Fig: Delay calculator wrapper. The blue arrow is taken if the delay calculator core currently expects write data for a previous write address request. Else, the black arrow is taken. Additionally, the red arrow gives the number of write data for the potentially incoming write address request</figcaption>
+</figure>
+
+#### Address request slots
+
+_Slots_ are used to store address request metadata, as well as the status of the elementary requests in the bursts.
+Write and read slots are disjoint, because of the chronology of write and read elementary burst request arrival:
+
+- Read elementary burst requests all arrive at once: when the read address request is opened.
+- Write elementary burst requests may arrive progressively, when write data requests arrive later than the corresponding write address request.
+
+_Write slots_ are memory structures defined as follows:
+
+- Valid (_v_): Is set iff the slot is currently occupied.
+- Internal identifier (_iid_): Contains the identifier that will be used to identify the extended cell where to release some response.
+- Address (_addr_): Contains the base address of the address request.
+  It is used to estimate the delay caused by the requests contained in the burst.
+- Burst size (_burst_size_): Contains the burst size of the address request.
+  It is used to calculate the address of the subsequent.
+  It may also be used in the future for improved delay estimation.
+- Data valid (_data_v_): Contains one bit per elementary burst request.
+  Each bit is unset iff the corresponding elementary burst request is awaited.
+- Memory pending (_mem_pending_): Contains one bit per elementary burst request.
+  Each bit is set iff there is currently a simulated memory operation pending.
+- Memory done (_mem_done_): Contains one bit per elementary burst request.
+  Each bit is set iff the corresponding simulated memory operation is done, or this is an entry beyond the burst length.
+
+_Read slots_ are defined identically, except that:
+
+- There is no _data_v_ field, since elementary burst read requests arrive simultaneously with the read address request.
+- There internal identifier for read and write address requests do not need to be of equal width, which depends on the maximal number of pending requests.
+
+The burst length does not need to be stored explicitly, as the _data_v_ and _mem_done_ bit arrays are set accordingly during the slot allocation.
+
+#### Delay estimation
+
+The delay estimation is performed using a decrementing counter (_rank_delay_cnt_) per rank (currently, only one rank is supported).
+When a simulated memory operation starts, the decrementing counter is set to a value corresponding to the situation:
+
+1. If this is a row hit.
```

Comment:
It has not  yet been discussed what a row is and how the addresses are mapped to ranks and rows etc.

Reply:
Mixed the correction with a previous comment!